internal/manager/run.go

```diff
@@ -78,11 +78,6 @@ func New(
 
 	gateway.SetControllerName(gatewayapi.GatewayController(c.GatewayAPIControllerName))
 
-	setupLog.Info("Getting enabled options and features")
-	featureGates, err := featuregates.New(setupLog, c.FeatureGates)
-	if err != nil {
-		return nil, fmt.Errorf("failed to configure feature gates: %w", err)
-	}
```

Comment:
The rest of PR is basically dedicated to getting rid of this. To not have any hidden config adjustments during the creation of a mangager